### PR TITLE
cmd/snap-repair:  start of Runner, implement first pass of Peek and Fetch

### PR DESCRIPTION
**Pull request by pedronis**
left TODO for later branches: 
 * use a time for TLS cert checks with a lower bound
 * wrapping logic to decide when to fallback to HTTP

### CODE REVIEW

File: cmd/snap-repair/runner_test.go [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+script
+
+
+AXNpZw==
+`
+	testHeadersResp = `{"headers":
+{"architectures":["amd64","arm64"],"authority-id":"canonical","body-length":"7","brand-id":"canonical","models":["xyz/frobinator"],"repair-id":"2","series":["16"],"sign-key-sha3-384":"KPIl7M4vQ9d4AUjkoU41TGAwtOMLc_bWUCeW8AvdRWD4_xcP60Oo4ABsFNo6BtXj","timestamp":"2017-03-30T12:22:16Z","type":"repair"}}`
+)
+
+func MustParseURL(s string) *url.URL {
```
Comment (pedronis) [inline cmd/snap-repair/runner_test.go]: no, just didn't think about that

File: cmd/snap-repair/runner.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+var ErrRepairNotFound = errors.New("repair not found")
+
+var (
+	maxRepairScriptSize = 24 * 1024 * 1024
+)
+
+// Fetch retrieves a stream with the repair with the given ids and any auxiliary assertions.
+func (run *Runner) Fetch(brandID, repairID string) (r []asserts.Assertion, err error) {
+	u, err := run.BaseURL.Parse(fmt.Sprintf("repairs/%s/%s", brandID, repairID))
```
Comment (pedronis) [inline cmd/snap-repair/runner.go]: for a while brandID will be a constant and repairID will be computed as the repr of an integer inside snap-repair, if we are worried about those at some point we should probably check higher than here?

Comment (chipaca) [inline cmd/snap-repair/runner.go]: I don't think we need worry, but thought it prudent to ask.

Comment (pedronis) [inline cmd/snap-repair/runner.go]: we could check that they have no "/"  in them if we wanted, that's a strong contraint on them anyway, it applies to anything that is used as part of the primary key of any assertion

**Pull request merged**